bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java

```diff
@@ -0,0 +1,520 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.heos.internal.HeosChannelHandlerFactory;
+import org.openhab.binding.heos.internal.HeosChannelManager;
+import org.openhab.binding.heos.internal.action.HeosActions;
+import org.openhab.binding.heos.internal.api.HeosFacade;
+import org.openhab.binding.heos.internal.api.HeosSystem;
+import org.openhab.binding.heos.internal.configuration.BridgeConfiguration;
+import org.openhab.binding.heos.internal.discovery.HeosPlayerDiscoveryListener;
+import org.openhab.binding.heos.internal.exception.HeosNotConnectedException;
+import org.openhab.binding.heos.internal.exception.HeosNotFoundException;
+import org.openhab.binding.heos.internal.json.dto.HeosError;
+import org.openhab.binding.heos.internal.json.dto.HeosEvent;
+import org.openhab.binding.heos.internal.json.dto.HeosEventObject;
+import org.openhab.binding.heos.internal.json.dto.HeosResponseObject;
+import org.openhab.binding.heos.internal.json.payload.Group;
+import org.openhab.binding.heos.internal.json.payload.Media;
+import org.openhab.binding.heos.internal.json.payload.Player;
+import org.openhab.binding.heos.internal.resources.HeosEventListener;
+import org.openhab.binding.heos.internal.resources.Telnet;
+import org.openhab.binding.heos.internal.resources.Telnet.ReadException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeosBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Johannes Einig - Initial contribution
+ */
+@NonNullByDefault
+public class HeosBridgeHandler extends BaseBridgeHandler implements HeosEventListener {
+    private final Logger logger = LoggerFactory.getLogger(HeosBridgeHandler.class);
+
+    private static final int HEOS_PORT = 1255;
+
+    private final List<HeosPlayerDiscoveryListener> playerDiscoveryList = new CopyOnWriteArrayList<>();
+    private final HeosChannelManager channelManager = new HeosChannelManager(this);
+    private final HeosChannelHandlerFactory channelHandlerFactory;
+
+    private final Map<String, HeosGroupHandler> groupHandlerMap = new ConcurrentHashMap<>();
+    private final Map<String, String> hashToGidMap = new ConcurrentHashMap<>();
+
+    private List<String[]> selectedPlayerList = new CopyOnWriteArrayList<>();
+
+    private @Nullable Future<?> startupFuture;
+    private List<Future<?>> childHandlerInitializedFutures = new ArrayList<>();
```

Comment:
Better safe than sorry.
```suggestion
    private List<Future<?>> childHandlerInitializedFutures = new CopyOnWriteArrayList<>();
```

Reply:
Will commit it myself, will also make the field final.